test/unit-test/mqtt_agent_command_functions_utest.c

```diff
@@ -73,3 +178,298 @@ int suiteTearDown( int numFailures )
+    mqttStatus = MQTTAgentCommand_Ping( &mqttAgentContext, pUnusedArg, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_TRUE( returnFlags.runProcessLoop );
+}
+
+/**
+ * @brief Test MQTTAgentCommand_Connect() success case.
+ */
+void test_MQTTAgentCommand_Connect( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    MQTTAgentConnectArgs_t connectInfo = { 0 };
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    MQTT_Connect_IgnoreAndReturn( MQTTSuccess );
+    MQTTAgent_ResumeSession_IgnoreAndReturn( MQTTSuccess );
+
+    mqttStatus = MQTTAgentCommand_Connect( &mqttAgentContext, &connectInfo, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_EQUAL( 0, returnFlags.packetId );
+}
+
+/**
+ * @brief Test MQTTAgentCommand_Connect() failure case.
+ */
+void test_MQTTAgentCommand_Connect_failure( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    MQTTAgentConnectArgs_t connectInfo = { 0 };
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    MQTTStatus_t mqttStatus;
+
+    MQTT_Connect_IgnoreAndReturn( MQTTBadParameter );
+
+    mqttStatus = MQTTAgentCommand_Connect( &mqttAgentContext, &connectInfo, &returnFlags );
+
+    TEST_ASSERT_EQUAL( MQTTBadParameter, mqttStatus );
+    /* Ensure that returnFlags are set as intended. */
+    TEST_ASSERT_EQUAL( 0, returnFlags.packetId );
+}
+
+/**
+ * @brief Test that MQTTAgentCommand_terminate() works as intended.
+ */
+void test_MQTTAgentCommand_terminate( void )
+{
+    MQTTAgentContext_t mqttAgentContext = { 0 };
+    MQTTStatus_t mqttStatus;
+    void * pUnusedArg;
+    MQTTAgentCommandFuncReturns_t returnFlags = { 0 };
+    Command_t command = { 0 };
+    AgentMessageInterface_t messageInterface = { 0 };
+
+    messageInterface.pMsgCtx = &globalMessageContext;
+    messageInterface.send = stubSend;
+    messageInterface.recv = stubReceive;
+    messageInterface.releaseCommand = stubReleaseCommand;
+    messageInterface.getCommand = stubGetCommand;
+
+    command.pCommandCompleteCallback = stubCompletionCallback;
+    mqttAgentContext.agentInterface = messageInterface;
+    mqttAgentContext.agentInterface.pMsgCtx->pSentCommand = &command;
+
+    mqttAgentContext.pPendingAcks[ 0 ].packetId = 1U;
+    mqttAgentContext.pPendingAcks[ 0 ].pOriginalCommand = &command;
+
+    mqttStatus = mqttStatus = MQTTAgentCommand_Terminate( &mqttAgentContext, pUnusedArg, &returnFlags );
+
+
+    TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
+
+    /* Ensure that callback is invoked. */
+    TEST_ASSERT_EQUAL( 2, commandCompleteCallbackCount );
+
+    /* Ensure that acknowledgment is cleared. */
+    TEST_ASSERT_EQUAL( 0, mqttAgentContext.pPendingAcks[ 0 ].packetId );
+    TEST_ASSERT_EQUAL( NULL, mqttAgentContext.pPendingAcks[ 0 ].pOriginalCommand );
+
```

Comment:
I believe both commands ought to be released. If not just add it here and let the test fail. It should pass once the bug is fixed:
```suggestion
    TEST_ASSERT_EQUAL( 2, commandReleaseCallCount );
```